cms/api.py

```diff
@@ -448,7 +448,12 @@ def __init__(self, user):
     request = FakeRequest(user)
     if not page.has_publish_permission(request):
         raise PermissionDenied()
-    page.publish(language)
+    # Set the current_user to have the page's changed_by
+    # attribute set correctly.
+    # 'user' is a user object, but current_user() just wants the username. Note
+    # that not all user models have `username`.
+    with current_user(unicode(user)):
```

Comment:
Check User.get_username() ( https://docs.djangoproject.com/en/1.7/ref/contrib/auth/#django.contrib.auth.models.User.get_username) which is guaranteed to exist on any user model